include/libdnf5/rpm/transaction_callbacks.hpp

```diff
@@ -92,6 +92,14 @@ class LIBDNF_API TransactionCallbacks {
     virtual void script_error(const TransactionItem * item, Nevra nevra, ScriptType type, uint64_t return_code);
     virtual void script_start(const TransactionItem * item, Nevra nevra, ScriptType type);
     virtual void script_stop(const TransactionItem * item, Nevra nevra, ScriptType type, uint64_t return_code);
+    /// Called after the scriptlet finished
+    /// @param item Transaction item
+    /// @param nevra NEVRA of the package the scriptlet belongs to
+    /// @param type Type of the scriptlet
+    /// @param return_code Return code of the scriptlet: OK, non-critical error, error
+    /// @param output Output produced by the scriptlet
+    virtual void script_output(
```

Comment:
PR inserts a new virtual method into the public callback class. This does not break the API.

However, adding a virtual function will definitely change the `vtable` and how it changes depends on the compiler.
**The result will be a broken ABI. And in a very bad way.** An application using a library with an added or missing virtual callback method will run and then crash during the callback function call.

If we want to keep ABI backward compatibility, we need to do it differently.

**Example solution:**
Instead of adding a new callback `script_output` (a virtual method), add a non-virtual API method `libdnf5::base::Transaction::get_last_scriptlet_output` that the library user will be able to call while servicing the existing callback `script_stop` .
This solution came to my mind first. Maybe there is a better place for a new non-virtual method. This is an example.